File: scala-route/src/main/scala/com/yangrui/spark/scala/SecondSort.scala
package com.yangrui.spark.scala

import org.apache.spark.{SparkConf, SparkContext}

object SecondSort {

  def main(args: Array[String]): Unit = {
    val conf = new SparkConf().setAppName("SecondSort").setMaster("local")
    val sc = new SparkContext(conf)
    val linesRDD=sc.textFile("C:\\Users\\admin\\Desktop\\123.txt");
    val secondSortKeyRDD=linesRDD.map(lines => {
      val lineSplits=lines.split(" ")
      val secondekey=new SecondSortKeyS(lineSplits(0).toInt,lineSplits(1).toInt)
      (secondekey,lines)
    })
    val sortRDD=secondSortKeyRDD.sortByKey(false)
    val afterRDD=sortRDD.map(tuple => tuple._2)
    afterRDD.foreach(line => println(line))

    sc.stop()
  }

}

File: scala-route/src/main/java/com/yangrui/spark/SecondSort.java
package com.yangrui.spark;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaPairRDD;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.Function;
import org.apache.spark.api.java.function.PairFunction;
import org.apache.spark.api.java.function.VoidFunction;
import scala.Tuple2;

public class SecondSort {

    public static void main(String[] args) {
        SparkConf sparkConf = new SparkConf().setAppName("SecondSort").setMaster("local");
        JavaSparkContext javaSparkContext = new JavaSparkContext(sparkConf);
        JavaRDD<String> lineRDD = javaSparkContext.textFile("C:\\Users\\admin\\Desktop\\123.txt");
        JavaPairRDD<SecondSortKey, String> secondSortKeyStringJavaPairRDD = lineRDD.mapToPair(new PairFunction<String, SecondSortKey, String>() {
            @Override
            public Tuple2<SecondSortKey, String> call(String s) throws Exception {
                String[] splits = s.split(" ");
                SecondSortKey ket = new SecondSortKey(Integer.parseInt(splits[0]), Integer.parseInt(splits[1]));
                return new Tuple2<SecondSortKey, String>(ket, s);
            }
        });
        JavaPairRDD<SecondSortKey, String> secondSortKeyStringJavaPairRDD1 = secondSortKeyStringJavaPairRDD.sortByKey(false);
        JavaRDD<String> map = secondSortKeyStringJavaPairRDD1.map(new Function<Tuple2<SecondSortKey, String>, String>() {
            @Override
            public String call(Tuple2<SecondSortKey, String> v1) throws Exception {
                return v1._2;
            }
        });
        map.foreach(new VoidFunction<String>() {
            @Override
            public void call(String s) throws Exception {
                System.out.println(s);
            }
        });
        javaSparkContext.close();
    }
}
